File: DotNetTreeShadows/Models/PieceType.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace dotnet_tree_shadows.Models {
    [JsonConverter( typeof( StringEnumConverter ) )]
    public enum PieceType {
        Seed,
        SmallTree,
        MediumTree,
        LargeTree
    }
}

File: DotNetTreeShadows/client/src/store/board/types/pieceType.ts
export enum PieceType {
    Seed,
    SmallTree,
    MediumTree,
    LargeTree
}

export function pieceTypeName(pieceType:PieceType, long?:boolean) {
    switch (pieceType) {
        case PieceType.Seed:
            return "Seed";
        case PieceType.SmallTree:
            return long ? "Small Tree" : "Small";
        case PieceType.MediumTree:
            return long
                   ? "Medium Tree"
                   :"Medium"
        case PieceType.LargeTree:
            return long
                   ? "Large Tree"
                   :"Large"
    }
}
